### PR TITLE
Fix grading period filter where count is close to paging limit

### PR DESCRIPTION
**Pull request by bluwave**
refs: MBL-13837
affects: Student
release note: none

Test plan:
1. use ticket to test https://instructure.atlassian.net/browse/MBL-13837
2. also, assignment list where there are many assignments and/or assignment groups
3. https://instructure.atlassian.net/browse/MBL-13585 this has a good account w/ mucho assignments to use to test

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-13837](https://instructure.atlassian.net/browse/MBL-13837)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.62% | 93.62% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

File: Student/Student/Assignments/AssignmentList/AssignmentListViewController.swift [hunk 1]
```diff
@@ -129,21 +134,23 @@ class AssignmentListViewController: UIViewController, ColoredNavViewProtocol, Er
         for ( index, g ) in groups.enumerated() {
             let groupID = g.id.value
             var existing = assignments[groupID] ?? []
+            let ass = g.assignments.map { $0.name }
+            print("assignments: \(ass)")
```
Comment (armstrongnate) [inline Student/Student/Assignments/AssignmentList/AssignmentListViewController.swift]: Oops!

File: Student/Student/Assignments/AssignmentList/AssignmentListViewController.swift [hunk 2]
```diff
@@ -156,7 +163,13 @@ class AssignmentListViewController: UIViewController, ColoredNavViewProtocol, Er
             self?.updateFilterButton()
             self?.selectFirstCellOnIpad()
             self?.loading = false
+            let maxPrefetch = 2
+            if self?.prefetchCount ?? 0 < maxPrefetch {
+                self?.prefetchCount +=  1
+                if self?.pagingCursor != nil { self?.fetchData() }
+            }
```
Comment (armstrongnate) [inline Student/Student/Assignments/AssignmentList/AssignmentListViewController.swift]: Help me understand why it's prefetching two? Will it not fetch as needed? What happens when there is a 3rd cursor?

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1021436&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

File: Student/Student/Assignments/AssignmentList/AssignmentListViewController.swift (hunk 2 above)
Comment (bluwave) [reply Student/Student/Assignments/AssignmentList/AssignmentListViewController.swift]: The page size is 10 on the simulator 11 rows can fit without scrolling.  It only pages when scrolled in the method using the `UITableViewDataSourcePrefetching` in the ```func tableView(_ tableView: UITableView, prefetchRowsAt indexPaths: [IndexPath])``` is called.

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1021439&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1021438&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1021442&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1021444&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1021450&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Pull request merged**